http/http_client.hpp

```diff
@@ -714,6 +710,18 @@ class ConnectionPool : public std::enable_shared_from_this<ConnectionPool>
             return;
         }
 
+        if (requestQueue.size() == maxRequestQueueSize)
```

Comment:
Will this part code be always executed everytime `requestQueue` needs to be changed?
Or is there any case that `requestQueue` may be changed more than `1` at time?

Perhaps would this be more safer?

```
if (requestQueue.size() >= maxRequestQueueSize)
```